prov/rxm/src/rxm_cq.c

```diff
@@ -161,7 +161,8 @@ static int rxm_finish_recv(struct rxm_rx_buf *rx_buf, size_t done_len)
 		if (ret)
 			return ret;
 	} else {
-		if (rx_buf->recv_entry->flags & FI_COMPLETION) {
+		if (rx_buf->recv_entry->flags & FI_COMPLETION ||
+		    rx_buf->ep->rxm_info->mode & FI_BUFFERED_RECV) {
```

Comment:
I think we should check for FI_CLAIM flag as well.

Also, I see a potential issue that probably applies only for SAR. If the user of buffered recv discards a SAR message but posts the recv with FI_DISCARD after RxM receives all the segments and is about to write the completion, the recv_entry would not have either FI_DISCARD or FI_COMPLETION flag set. It seems we have to en-queue it to some sort of pending list.